### PR TITLE
Module11

### PR DESCRIPTION
**Pull request by Arpitha-debug**
Please make sure you are raising this PR against your own repository and not the original. If it says https://github.com/CorndelWithSoftwire in the search bar right now, you are in the wrong place!

Please also don't forget to submit the PR's url to Aptem afterwards (by clicking on the corresponding exercise component, pasting the url in the textbox that appears and then pressing the finish button).

### CODE REVIEW

File: .github/workflows/ci.yml
Comment (JackMeadDev) [inline .github/workflows/ci.yml]: I think that we changed from using a specific environment to just using the repository level secrets, so we should be able to remove this now

File: .github/workflows/ci.yml
Comment (JackMeadDev) [inline .github/workflows/ci.yml]: Let's delete rather than commenting out!

File: .github/workflows/ci.yml
Comment (JackMeadDev) [inline .github/workflows/ci.yml]: We should still be running your unit tests (at least!) as part of this pipeline - switching them off for the sake of testing your deployment is reasonable but we should make sure we have them back before merging

File: todo_app/SeleniumTest/Tests/test_e2e.py
Comment (JackMeadDev) [inline todo_app/SeleniumTest/Tests/test_e2e.py:1]: If we're disabling tests, we should either:
* Mark them to be skipped with `@pytest.mark.skip(reason="these need updating")` so that future users can see what the reason is and reason when they should be switched on
* Or delete them entirely!

File: todo_app/data/mongo.py
Comment (JackMeadDev) [inline todo_app/data/mongo.py:4]: Nice restructuring to pull this out

File: todo_app/data/mongo.py
Comment (JackMeadDev) [inline todo_app/data/mongo.py]: I don't think you use this value any more (since we moved to refreshing it each time through the `todos_collection` since otherwise it wouldn't update) so we can get rid of this

File: todo_app/tests/unit/mockmongotest.py
Comment (JackMeadDev) [inline todo_app/tests/unit/mockmongotest.py]: Why is there random pipeline code in here?

File: todo_app/tests/unit/test_integration.py
Comment (JackMeadDev) [inline todo_app/tests/unit/test_integration.py]: Unless we're going to fix this up later, let's just get rid of it

File: todo_app/app.py
Comment (JackMeadDev) [inline todo_app/app.py:21]: The code here suggests that your app should be checking authentication, but I can access your app in Azure without it; do you know why that is? We should ensure the latest build of your code is actually running in Azure

**Review by JackMeadDev — APPROVED** [review]
This looks good in that I can see the app is successfully deploying to Azure and is successfully running there, but it's still a bit confused; for example your `app.py` is written to manage authentication, but the deployed app isn't doing that and looks to still be using Trello instead of Mongo. I'm happy for you to merge this as is, but once M8/9/10/11 are all merged we should ensure that the latest version of the app is the one that Azure is running, and that whatever tests are still valid (potentially just the ViewModel unit tests) are still being run by your CI pipeline

File: .github/workflows/ci.yml
Comment (Arpitha-debug) [reply .github/workflows/ci.yml]: Done

File: .github/workflows/ci.yml
Comment (Arpitha-debug) [reply .github/workflows/ci.yml]: ok

File: todo_app/data/mongo.py
Comment (Arpitha-debug) [reply todo_app/data/mongo.py]: deleted

File: todo_app/tests/unit/mockmongotest.py
Comment (Arpitha-debug) [reply todo_app/tests/unit/mockmongotest.py]: deleted

File: todo_app/tests/unit/test_integration.py
Comment (Arpitha-debug) [reply todo_app/tests/unit/test_integration.py]: deleted

File: .github/workflows/ci.yml
Comment (Arpitha-debug) [reply .github/workflows/ci.yml]: added mockmongo tests

File: todo_app/SeleniumTest/Tests/test_e2e.py
Comment (Arpitha-debug) [reply todo_app/SeleniumTest/Tests/test_e2e.py:1]: skipped integration tests

File: todo_app/data/mongo.py
Comment (Arpitha-debug) [reply todo_app/data/mongo.py]: readded for mockmongo tests

File: todo_app/app.py
Comment (Arpitha-debug) [reply todo_app/app.py:21]: Added another OAuth for Azure

**Pull request merged**
